File: programmers/Courses/알고리즘-문제-해설/3.cpp
//
// Created by habi on 2023-08-01.
//
// Problem
// https://school.programmers.co.kr/learn/courses/18/lessons/1878
//

#include <iostream>
#include <vector>
using namespace std;

vector<int> solution(vector<vector<int>> v) {
  vector<int> ans = {0, 0};

  /*
   * 중요한 개념
   * (A XOR A) XOR B = B
   * (A XOR B) XOR B = A
   */
  for (int i = 0; i < 3; i++) {
    ans[0] ^= v[i][0];
    ans[1] ^= v[i][1];
  }

  return ans;
}

File: programmers/Courses/알고리즘-문제-해설/3.py
#
# Created by habi on 2023-08-01.
#
# Problem
# https://school.programmers.co.kr/learn/courses/18/lessons/1878
#


def check(arr):
    if arr[0] == arr[1]:
        return arr[2]
    elif arr[0] == arr[2]:
        return arr[1]
    else:
        return arr[0]


def solution(v):
    answer = []
    vx = []
    vy = []

    for i in range(3):
        vx.append(v[i][0])
        vy.append(v[i][1])

    answer = [check(vx), check(vy)]

    return answer
